src/components/visualization/mermaidViz/mermaidViz.tsx

```diff
@@ -0,0 +1,48 @@
+import './mermaidViz.css'
+
+import Stack from '@mui/material/Stack'
+import Typography from '@mui/material/Typography'
+import useTheme from '@mui/system/useTheme'
+import mermaid from 'mermaid'
+import React, { useEffect, useRef } from 'react'
+
+import type { MermaidData } from '../../types'
+
+/** The `MermaidViz` component leverages [Mermaid.js](https://mermaid.js.org/) to create dynamic and interactive diagrams, including flowcharts, sequence diagrams, class diagrams, and more. It is ideal for visualizing complex data and processes in a clear and structured manner. */
+function MermaidViz(props: MermaidData) {
+  const chartRef = useRef(null)
+  const rusticTheme = useTheme()
+  const mermaidTheme = rusticTheme.palette.mode === 'dark' ? 'dark' : 'neutral'
+
+  useEffect(() => {
+    if (chartRef.current) {
+      mermaid.initialize({
+        theme: mermaidTheme,
+        themeVariables: { fontFamily: 'Inter' },
+        ...props.config,
+      })
+
+      mermaid.contentLoaded()
```

Comment:
I think we should be using `mermaid.run` instead. It allows more flexibility for the selectors and suppressing errors so that we can show custom error messages.

Refer - https://mermaid.js.org/config/usage.html#using-mermaid-run